File: CheckListBox LoadingRetrieving items from a Database/[C#][VB]-CheckListBox LoadingRetrieving items from a Database/C# and VB/CheckListBoxFromMS_Access/LanguageExtensionMethods/CheckedListBoxExtensions.vb
Public Module CheckedListBoxExtensions
    ''' <summary>
    ''' Find item, set check state
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="Text">text to locate case insensitive</param>
    ''' <param name="Checked">set check state to</param>
    ''' <remarks></remarks>
    <System.Diagnostics.DebuggerStepThrough()> _
    <System.Runtime.CompilerServices.Extension()> _
    Public Sub FindItemAndSetChecked(ByVal sender As CheckedListBox, ByVal Text As String, ByVal Checked As Boolean)
        Dim Result =
            (
                From this In sender.Items.Cast(Of String)() _
                .Select(Function(item, index) New With {.Item = item, .Index = index})
                Where this.Item.ToLower = Text.ToLower
            ).FirstOrDefault

        If Result IsNot Nothing Then
            sender.SetItemChecked(Result.Index, Checked)
        End If

    End Sub
End Module

File: CheckListBox LoadingRetrieving items from a Database/[C#][VB]-CheckListBox LoadingRetrieving items from a Database/C# and VB/CheckListBoxSimple_C_Sharp/LanguageExtensionMethods/CheckedListBoxExtensions.cs
using System.Linq;
using System.Windows.Forms;

public static class CheckedListBoxExtensions
{
    /// <summary>
    /// Find item, set check state
    /// </summary>
    /// <param name="sender"></param>
    /// <param name="Text">text to locate case insensitive</param>
    /// <param name="Checked">set check state to</param>
    /// <remarks></remarks>
    public static void FindItemAndSetChecked(this CheckedListBox sender, string Text, bool Checked)
    {
        // Result is declared with var as it's a anonymous type
        var Result =
            (
                from @this in sender.Items.Cast<string>().Select(
                    (item, index) => new
                        {
                            Item = item,
                            Index = index
                        }
                    )
                where @this.Item.ToLower() == Text.ToLower()
                select @this
            ).FirstOrDefault();

        if (Result != null)
        {
            sender.SetItemChecked(Result.Index, Checked);
        }
    }
}